### PR TITLE
feat: AWS GovCloud Config integration

### PR DESCRIPTION
**Pull request by dmurray-lacework**
AWS Gov Cloud Config integration

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: api/integrations_aws_gov_cloud.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+//         AccountID: "553453453",
+//         AccessKeyID: "AWS123abcAccessKeyID",
+//         SecretAccessKey: "AWS123abc123abcSecretAccessKey0000000000",
+//       },
+//     },
+//   )
+//
+//   client.Integrations.CreateAwsGovCloudCfg(aws)
+//
+func NewAwsGovCloudCfgIntegration(name string, iType integrationType, data AwsGovCloudIntegrationData) AwsGovCloudIntegration {
```
Comment (afiune) [inline api/integrations_aws_gov_cloud.go:45]: uhmmm, this is confusing, why are we passing an integration type if we are specifically saying that we are creating an `AWS_US_GOV_CFG` integration? I think this was a copy-paste from the main `NewAwsIntegration()` function, actually, did you try to use it instead?

Could we do something like:
```go
   client, err := api.NewClient("account")
   if err != nil {
     return err
   }

   aws := api.NewAwsIntegration("foo",
     api.AwsGovCloudCfgIntegration,
     api.AwsIntegrationData{
       GovCloudCredentials: api.AwsGovCloudCreds {},
     },
   )

   client.Integrations.CreateAws(aws)
```

File: cli/cmd/integration.go [hunk 2]
```diff
@@ -279,6 +280,8 @@ func promptCreateIntegration() error {
 		return createAwsConfigIntegration()
 	case "AWS CloudTrail":
 		return createAwsCloudTrailIntegration()
+	case "AWS Gov Cloud Config":
```
Comment (afiune) [inline cli/cmd/integration.go:283]: It is AWS GovCloud: https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/govcloud-differences.html
```suggestion
	case "AWS GovCloud Config":
```

File: cli/cmd/integration.go [hunk 3]
```diff
@@ -221,6 +221,7 @@ func promptCreateIntegration() error {
 				"Google Container Registry (GCR)",
 				"AWS Config",
 				"AWS CloudTrail",
+				"AWS Gov Cloud Config",
```
Comment (afiune) [inline cli/cmd/integration.go:224]:
```suggestion
				"AWS GovCloud Config",
```

File: api/integrations_aws_gov_cloud.go (hunk 1 above)
Comment (dmurray-lacework) [reply api/integrations_aws_gov_cloud.go:45]: I'm not sure Salim, would AwsIntegrationData then have to accept GovCloudCredentials type?

File: cli/cmd/integration_aws_gov_cloud.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+			Validate: survey.Required,
+		},
+		{
+			Name:     "access_key_id",
+			Prompt:   &survey.Input{Message: "Access Key ID: "},
+			Validate: survey.Required,
+		},
+		{
+			Name:     "secret_access_key",
+			Prompt:   &survey.Input{Message: "Secret Access Key: "},
```
Comment (afiune) [inline cli/cmd/integration_aws_gov_cloud.go:46]: This is a sensitive value. Make it a secret:
```suggestion
			Prompt:   &survey.Password{Message: "Secret Access Key: "},
```

File: api/integrations_aws.go [hunk 5]
```diff
@@ -146,6 +146,8 @@ type AwsIntegrationData struct {
 
 	// AwsAccountID is the AWS account that owns the IAM role credentials
 	AwsAccountID string `json:"AWS_ACCOUNT_ID,omitempty" mapstructure:"AWS_ACCOUNT_ID"`
+
+	GovCloudCredentials AwsGovCloudCreds `json:"ACCESS_KEY_CREDENTIALS,omitempty" mapstructure:"ACCESS_KEY_CREDENTIALS"`
```
Comment (afiune) [inline api/integrations_aws.go:150]: You will have to add `omitempty` to the `CROSS_ACCOUNT_CREDENTIALS ` field. If not,
we will have a Schema Validation failure. (I think)

**Review by afiune — APPROVED** [review]
would be nice to add more tests for this new integration type... 

tenor-194735931

File: api/integrations.go [hunk 6]
```diff
@@ -40,6 +40,12 @@ const (
 	// AWS CloudTrail integration type
 	AwsCloudTrailIntegration
 
+	// AWS Gov Cloud Config integration type
+	AwsGovCloudCfgIntegration
+
+	// AWS GovCloud CloudTrail integration type
+	AwsGovCloudCTIntegration
```
Comment (afiune) [inline api/integrations.go:47]: Shouldn't this be 🤔
```suggestion
	AwsGovCloudCloudTrailIntegration
```

File: cli/cmd/integration.go [hunk 7]
```diff
@@ -221,6 +221,8 @@ func promptCreateIntegration() error {
 				"Google Container Registry (GCR)",
 				"AWS Config",
 				"AWS CloudTrail",
+				"AWS GovCloud Config",
+				"AWS GovCloud CloudTrail",
```
Comment (afiune) [inline cli/cmd/integration.go:225]: To match a bit how UI shows this "dropdown menu" should we call them like this?
```suggestion
				"AWS Config (US GovCloud)",
				"AWS CloudTrail (US GovCloud)",
```

Screen Shot 2021-05-26 at 11 43 05 AM

File: api/integrations_aws_test.go [hunk 8]
```diff
@@ -331,3 +494,58 @@ func singleAwsIntegration(id string) string {
...
+			"STATE": {
+				"ok": true,
+				"lastUpdatedTime": "2020-Mar-10 01:00:00 UTC",
+				"lastSuccessfulTime": "2020-Mar-10 01:00:00 UTC"
+			},
+			"IS_ORG": 0,
+			"DATA": {
+ 				"QUEUE_URL": "https://sqs.us-gov-west-1.amazonaws.com/123456789012/my_queue",
+				"ACCESS_KEY_CREDENTIALS": {
+          			"ACCOUNT_ID": "0123456789",
```
Comment (afiune) [inline api/integrations_aws_test.go:543]: nit
```suggestion
          				"ACCOUNT_ID": "0123456789",
```

File: api/integrations_aws_test.go [hunk 9]
```diff
@@ -331,3 +494,58 @@ func singleAwsIntegration(id string) string {
...
+			"ENABLED": 1,
+			"STATE": {
+				"ok": true,
+				"lastUpdatedTime": "2020-Mar-10 01:00:00 UTC",
+				"lastSuccessfulTime": "2020-Mar-10 01:00:00 UTC"
+			},
+			"IS_ORG": 0,
+			"DATA": {
+				"ACCESS_KEY_CREDENTIALS": {
+          			"ACCOUNT_ID": "0123456789",
```
Comment (afiune) [inline api/integrations_aws_test.go:515]:
```suggestion
          				"ACCOUNT_ID": "0123456789",
```

File: api/integrations_aws_test.go [hunk 10]
```diff
@@ -208,15 +351,15 @@ func TestIntegrationsUpdateAws(t *testing.T) {
...
 				ExternalID: "0123456789",
 			},
 		},
 	)
-	assert.Equal(t, "integration_name", data.Name, "GCP integration name mismatch")
-	assert.Equal(t, "AWS_CT_SQS", data.Type, "a new GCP integration should match its type")
-	assert.Equal(t, 1, data.Enabled, "a new GCP integration should be enabled")
+	assert.Equal(t, "integration_name", data.Name, "AWS integration name mismatch")
+	assert.Equal(t, "AWS_CT_SQS", data.Type, "a new AWS integration should match its type")
+	assert.Equal(t, 1, data.Enabled, "a new AWS integration should be enabled")
```
Comment (afiune) [inline api/integrations_aws_test.go:362]: thank you!

File: cli/cmd/integration.go [hunk 11]
```diff
@@ -279,6 +281,10 @@ func promptCreateIntegration() error {
 		return createAwsConfigIntegration()
 	case "AWS CloudTrail":
 		return createAwsCloudTrailIntegration()
+	case "AWS GovCloud Config":
+		return createAwsGovCloudConfigIntegration()
+	case "AWS GovCloud CloudTrail":
+		return createAwsGovCloudCTIntegration()
```
Comment (afiune) [inline cli/cmd/integration.go:287]: nit: If we change the Integration type, would be nice to change these functions too
```suggestion
		return createAwsGovCloudCloudTrailIntegration()
```

**Review by afiune — COMMENTED** [review]
That is it! 

tenor-24089157

File: api/integrations.go (hunk 6 above)
Comment (dmurray-lacework) [reply api/integrations.go:47]: Depends on how you feel about CloudCloud. I went with CT because I didn't like the look of it.

**Pull request merged**